eth/downloader/downloader.go

```diff
@@ -1337,26 +1360,46 @@ func (d *Downloader) processHeaders(origin uint64, td, ttd *big.Int, beaconMode
 
 				// In case of header only syncing, validate the chunk immediately
 				if mode == SnapSync || mode == LightSync {
+					// Although the received headers might be all valid, a legacy
+					// PoW/PoA sync must not accept post-merge headers. Make sure
+					// that any transition is rejected at this point.
+					var (
+						rejected []*types.Header
+						td       *big.Int
+					)
+					if !beaconMode && ttd != nil {
```

Comment:
for bsc,  ttd is nil forever
so these codes are useless, but keep them is ok, making code align with go-ethereum